R4R: Don't hardcode bondable denom

alessio opened the pull request:
- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [x] Added entries in `PENDING.md` with issue # 
- [x] rereviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

jaekwon commented:
This still has "stake" in many places, doesn't seem complete?  Also this will break 9000 and the genesis, so not merging for 9000.

jaekwon commented:
Will merge once replacing "stake" with e.g. "STAKE" in x/stake/types still passes tests.

jaekwon commented:
I've added a commit w/ more replacements and temporarily replaced DefaultBondDenom with "STAKE", but the tests don't pass yet.

NOTE: Can't merge this for 9001.

alessio commented:
Fixed now

On Mon, 12 Nov 2018, 01:46 Jae Kwon <notifications@github.com wrote:

> I've added a commit w/ more replacements and temporarily replaced
> DefaultBondDenom with "STAKE", but the tests don't pass yet.
>
> NOTE: Can't merge this for 9001.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cosmos/cosmos-sdk/pull/2752#issuecomment-437727879>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAN_7Nw_kXkkn-5fzeEkgo3W_o3px42Yks5uuNMBgaJpZM4YXO8l>
> .
>

alexanderbez commented on x/auth/client/txbuilder/txbuilder_test.go:
```diff
@@ -55,7 +55,7 @@ func TestTxBuilderBuild(t *testing.T) {
 				Memo:          "hello",
 				Msgs:          defaultMsg,
-				Fee:           auth.NewStdFee(100, sdk.NewCoin("steak", sdk.NewInt(1))),
+				Fee:           auth.NewStdFee(100, sdk.NewCoin("stake", sdk.NewInt(1))),
```
Should we update this too?

alessio replied:
That shouldn't matter too much. One can put anything instead of stake

alexanderbez replied:
Just for consistency I suppose. Every other test was updated.

alessio replied:
Done

jackzampolin commented:
The PR as written passes tests for me, but if I go to `x/stake/types/params.go` and change the `DefaultBondDenom` to `stajake`, `make test` returns some errors:

```
--- FAIL: TestMsgSubmitProposal (0.00s)
        Error Trace:    msgs_test.go:46
        Error:      	Received unexpected error:
                    	ERROR:
                    	Codespace: 1
                    	Code: 11
                    	Message: "1000stajake,10000foo"
        Test:       	TestMsgSubmitProposal
        Messages:   	test: 9
--- FAIL: TestMsgDeposit (0.00s)
        Error Trace:    msgs_test.go:72
        Error:      	Received unexpected error:
                    	ERROR:
                    	Codespace: 1
                    	Code: 11
                    	Message: "1000stajake,10000foo"
        Test:       	TestMsgDeposit
        Messages:   	test: 4
```

alessio commented:
@jackzampolin try again now
